cassovary-core/src/test/scala/com/twitter/cassovary/util/io/ListOfEdgesGraphReaderSpec.scala

```diff
@@ -53,6 +56,37 @@ class ListOfEdgesGraphReaderSpec extends WordSpec with GraphBehaviours[Node] wit
       idReader = _.toLong).toSharedArrayBasedDirectedGraph()
   }
 
+  trait GraphWithStringIdsWithSortReader {
```

Comment:
there is a lot of duplication of code here because of the various cases to cover. Would be better to not instantiate named traits but classes with different variables e.g., class Graph(fileName: String, sortNeighbors, removeDuplicates) and then do the same tests for all the combinations of the variables sortNeighbors and removeDuplicates